pkg/controller/common.go

```diff
@@ -627,23 +629,27 @@ func getOperationRequeueError(desired string, obj client.Object) error {
 	return status.Errorf(codes.Aborted, "requeueing %s operation because another operation is already pending on %v (%s)", desired, reflect.TypeOf(obj), obj.GetName())
 }
 
-func reconcileReturnOnError(obj interface{}, operationType string, err error) (reconcile.Result, error) {
-	var errMsg string
-	if obj != nil {
-		var objName string
-		switch target := obj.(type) {
-		case *v1alpha1.AzVolume:
-			objName = target.Name
-		case *v1alpha1.AzVolumeAttachment:
-			objName = target.Name
+func reconcileReturnOnError(obj runtime.Object, operationType string, err error, retryInfo *retryInfo) (reconcile.Result, error) {
+	var (
+		requeue    bool = status.Code(err) != codes.FailedPrecondition
+		retryAfter time.Duration
+	)
+
+	if meta, metaErr := meta.Accessor(obj); metaErr == nil {
+		objectName := meta.GetName()
+		objectType := reflect.TypeOf(obj)
+		if !requeue {
+			klog.Errorf("failed to %s %v (%s) with no retry: %v", operationType, objectType, objectName, err)
```

Comment:
Do we want to remove an existing entry from `retryInfo` (if there were one added from previous reconciliation) if we encounter an error that we wish not requeue on?

Reply:
Ah, yes. Good catch!